lib/service/service.go

```diff
@@ -1998,11 +2013,101 @@ func (process *TeleportProcess) initUploaderService(accessPoint auth.AccessPoint
 	return nil
 }
 
+// initMetricsService starts the metrics service currently serving metrics for
+// prometheus consumption
+func (process *TeleportProcess) initMetricsService() error {
+	mux := http.NewServeMux()
+	mux.Handle("/metrics", promhttp.Handler())
```

Comment:
That was the original plan but the goal is to enable mtls on the metrics endpoint only and leave the others as is.
That can be theoretically possible if we trigger a tls renegotiation on the metrics endpoint, but it's both not supported by prometheus and also deprecated in tls 1.3

Reply:
@awly The suggestion to use /healthz and /readyz were not part of the RFD. I don't believe it's guarenteed that a customer would want this on mTLS for healthz/readyz checks, and systems like kubernetes to my knowledge don't support mtls in healthz/readyz checks. If the option is wanted to expose those checks over mtls and not with http checks, my 2 cents is those should be individually configurable.

Reply:
OK, I can understand leaving out `/healthz` and `/readyz`, but all the `/debug` ones seem logically closer to metrics and should be included here when both `--diag-addr` and `metrics_service` are enabled.

FWIW, I think it will be extremely confusing for on-prem customers to understand the difference between web UI, metrics and diag ports. I should've thought of this during RFD review, my bad.
Let's proceed according to the RFD, but please document in a comment and `/docs/` the difference between all these endpoints and ports.